### PR TITLE
Fixes #305 - incorrect ui window selection in Neovim

### LINKED ISSUE #305: Vdebug calculates source window incorrectly in Neovim
When using neovim and trying to debug something that's further away from the first windows, it breaks the layout, or even throws an exception. Here I had 6 tabs opened and 7th was my source file with breakpoint:
Vim:
```
- [Info] {Sat 07 2017 12:58:28} Searching for win by name /home/vagrant/test.php
- [Info] {Sat 07 2017 12:58:28} Win 1, name /home/vagrant/test.php
- [Info] {Sat 07 2017 12:58:28} Returning window number 1
- [Info] {Sat 07 2017 12:58:28} Found window 1, activating
- [Info] {Sat 07 2017 12:58:28} Found connection from ('127.0.0.1', 37240)
....
```
Neovim:
```
- [Info] {Sat 07 2017 12:52:22} Searching for win by name /home/vagrant/test.php
- [Info] {Sat 07 2017 12:52:22} Win 1, name 
- [Info] {Sat 07 2017 12:52:22} Win 2, name 
- [Info] {Sat 07 2017 12:52:22} Win 3, name 
- [Info] {Sat 07 2017 12:52:22} Win 4, name 
- [Info] {Sat 07 2017 12:52:22} Win 5, name 
- [Info] {Sat 07 2017 12:52:22} Win 6, name 
- [Info] {Sat 07 2017 12:52:22} Win 7, name /home/vagrant/test.php
- [Info] {Sat 07 2017 12:52:22} Returning window number 7
- [Info] {Sat 07 2017 12:52:22} Found window 7, activating
- [Info] {Sat 07 2017 12:52:22} Closing the connection
- [Debug] {Sat 07 2017 12:52:22} Command: stop -i 1
- [Debug] {Sat 07 2017 12:52:22} Response: <?xml version="1.0" encoding="iso-8859-1"?>
<response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="http://xdebug.org/dbgp/xdebug" command="stop" transaction_id="1" status="stopped" reason="ok"></response>
- [Debug] {Sat 07 2017 12:52:22} Closing the socket
- [ERROR] {Sat 07 2017 12:52:22} An error occured: <class 'neovim.api.nvim.NvimError'>
Traceback (most recent call last):
  File "/home/vagrant/.vim/bundle/vdebug/plugin/python/start_vdebug.py", line 31, in run
    self.runner.run()
  File "/home/vagrant/.vim/vdebug/plugin/python/vdebug/runner.py", line 184, in run
    self.open()
  File "/home/vagrant/.vim/vdebug/plugin/python/vdebug/runner.py", line 48, in open
    self.ui.open()
  File "/home/vagrant/.vim/vdebug/plugin/python/vdebug/ui/vimui.py", line 80, in open
    raise e
NvimError: Vim(wincmd):E16: Invalid range

```
It seems that vim-python manages windows differently in vim and neovim. In vim it is managed by-tab, in neovim it is managed globally

`vim -c "vsplit" -c "tabnew" -c "py import vim"`
```
:py print len(vim.windows)
```
Vim outputs 1
Neovim outputs 3

I hacked some sort of a PR, maybe you have more sophisticated idea what is going on and how to fix better.

### PR DESCRIPTION
**Pull request by reafle**
Vim and Neovim seem to manage windows differently.
This patch removes redundant window search for source window
and assumes that the source window is always nr. 1.

### CODE REVIEW

**Comment by reafle** [comment]
fixes #305

**Comment by cprn** [comment]
Just to quickly confirm assuming window number is 1 solves the issue in all my personal use cases (I know it's not ideal).

Also, I think Travis wouldn't complain if you left everything and just fixed line 75 of the original file where `winnr` is passed to `SourceWindow()`:

```diff
-self.sourcewin = SourceWindow(self,winnr)
+self.sourcewin = SourceWindow(self,1)
```

Build is failing because you've removed methods covered by tests:

**Comment by reafle** [comment]
I'm not too keen on leaving unused routines in the codebase.

Also travis complains about every PR to a master, because its configured for the v2-interation branch, you should check out the message

**Comment by lucc** [comment]
Still the removing of unused methods is logically independent from the fix for neovim and could go into a separate commit. Then it could be tested and merged individually (whatever the maintainer decides).

**Comment by reafle** [comment]
@lucc are the two commits usually done in one PR ? Or two PRs, one fixes the functionality, the other cleans unused code?

**Comment by lucc** [comment]
That all depends on your style or the style of the project/maintainer.

I tend to think like this:
* Every commit is a logical unit, something that can easily described in the message. If your message says "do this ... AND do that ... And by the way also do something else ... AND some cleanup" then the commit should be split :)
* I try to make the `git log` output look like a cleaned up comprehensible story about what I did during development. I tend to reorder and clean up commits until I like it.
* If possible the tests should pass after *every commit*. Helps with `git bisect`.
* A PR is a proposal for the team or maintainer. So I bundle some commits together and tell them: "Here take that". If my commit messages "tell a nice story" I hope they understand what I did and why and how.

But the question what to bundle together is very dependent on the team I work with, the amount of work I did in one series of patches and also the "closeness" of the changes. 

In this case I would leave both commits in this PR here.

**Comment by BlackIkeEagle** [comment]
tested an applied in v2-integration, thanks

**Comment by reafle** [comment]
@lucc thanks for your insights on commits and pull requests, lots of good thoughts there. 

Personally, in my daily work, I tend not to care how many commits are in the pull request, since all I care about (as a project maintainer) is for the new feature/a hotfix/whatever that makes a concrete impact on the outcome of the merge. So we usually go 1 PR = 1 commit, that should be pretty minimalistic. So for example, if you add a new feature AND refactor some old code, it is better to make 2 PRs (with 2 commits) than 1 PR with 2 commits.

In other words, if stuff needs some dedicated attention as a step in the development process, it goes into a PR. If stuff is an addition to the current task you are implementing, it can go hand in hand in the same commit. More PRs allow for more active integration process.

But that's just how we usually do it, your approach also has a nicer feel of a "story to tell".
